### ISSUE TITLE
Javascript error after Firefox v0.77 upgrade. 

### ISSUE THREAD

**Issue by vandertilt** [mozilla/geckodriver#1730]
## System

* Version: geckodriver-v0.24.0-win64
* Platform: Windows 10
* Firefox: v.0.77
* Selenium: Python 3.141.0

## Testcase
**NB**: *First ever issue post on GitHub, so excuse my "formatting blunders"*

* Problem:
   * driver.get_url("https://www.google.com") fails with javascript error, both headless and not.
   * does not fail for driver.get_url("https://www.mozilla.com")
   * Some sites work, some do not. But 100% consistent behaviour per site.
* Workaround:
   * roll back to Firefox v.0.76.0.1

## Log

1591219232836	Marionette	INFO	Stopped listening on port 51403
1591219344234	mozrunner::runner	INFO	Running command: "C:\\Program Files\\Mozilla Firefox\\firefox.exe" "-marionette" "-foreground" "-no-remote" "-profile" "C:\\Users\\Boostface\\AppData\\Local\\Temp\\rust_mozprofileiwn16I"
1591219345477	Marionette	INFO	Listening on port 51896
1591219345776	Marionette	WARN	TLS certificate errors will be ignored for this session
JavaScript error: resource://gre/modules/URIFixup.jsm, line 203: NS_ERROR_FAILURE: Should pass a non-null uri
1591219427042	Marionette	INFO	Stopped listening on port 51896
1591219544305	mozrunner::runner	INFO	Running command: "C:\\Program Files\\Mozilla Firefox\\firefox.exe" "-marionette" "-foreground" "-no-remote" "-profile" "C:\\Users\\Boostface\\AppData\\Local\\Temp\\rust_mozprofilehEkdjX"
1591219545683	Marionette	INFO	Listening on port 52103
1591219545916	Marionette	WARN	TLS certificate errors will be ignored for this session
JavaScript error: resource://gre/modules/URIFixup.jsm, line 203: NS_ERROR_FAILURE: Should pass a non-null uri

**Comment by whimboo** [comment]
Could you please attach a [trace-level log] from geckodriver? Read more about reporting actionable bugs in our [contribution guidelines].

[trace-level log]: https://firefox-source-docs.mozilla.org/testing/geckodriver/TraceLogs.html
[contribution guidelines]: https://firefox-source-docs.mozilla.org/testing/geckodriver/Bugs.html

**Comment by skanungo2007** [comment]
Facing the same issue with Selenium Java after upgrade to version 77. 
JavaScript error: resource://gre/modules/URIFixup.jsm, line 203: NS_ERROR_FAILURE: Should pass a non-null uri

**Comment by vandertilt** [comment]
Unfortunately I was unable to recreate the issue by upgrading to v0.77 again.
I'll make sure to collect the necessary trace-level logs in case the issue re-surfaces. Unless this was a wider spread issue, I assume it is only reasonable to close this issue for now?

**Comment by skanungo2007** [comment]
If you check the forums, you can see many others reported the same. Even ai
confirmed the same issue by installing version 77 on another system. Both
running Windows 10.

**Comment by whimboo** [comment]
@skanungo2007 which pages are failing for you?

**Comment by whimboo** [comment]
Also please provide a trace log as requested earlier.

**Comment by francogu** [comment]
```
2020-06-24 10:21:54.074  INFO 30131 --- [           main] c.c.s.i.WebDriverThreadLocalContainer    : No webdriver is bound to current thread: 1 - let's create a new webdriver
1592986914396	mozrunner::runner	INFO	Running command: "/usr/lib/firefox/firefox" "-marionette" "-foreground" "-no-remote" "-profile" "/tmp/rust_mozprofileMWB9zV"
1592986914396	geckodriver::marionette	DEBUG	Waiting 60s to connect to browser on 127.0.0.1:33139
1592986914849	Marionette	TRACE	Marionette enabled
1592986914902	Marionette	TRACE	Received observer notification toplevel-window-ready
1592986915740	Marionette	TRACE	Received observer notification marionette-startup-requested
1592986915740	Marionette	TRACE	Waiting until startup recorder finished recording startup scripts...
1592986915751	Marionette	TRACE	All scripts recorded.
1592986915751	Marionette	DEBUG	Setting recommended pref apz.content_response_timeout to 60000
1592986915751	Marionette	DEBUG	Setting recommended pref browser.contentblocking.introCount to 99
1592986915751	Marionette	DEBUG	Setting recommended pref browser.download.panel.shown to true
1592986915751	Marionette	DEBUG	Setting recommended pref browser.newtabpage.enabled to false
1592986915752	Marionette	DEBUG	Setting recommended pref browser.pagethumbnails.capturing_disabled to true
1592986915752	Marionette	DEBUG	Setting recommended pref browser.search.update to false
1592986915752	Marionette	DEBUG	Setting recommended pref browser.tabs.disableBackgroundZombification to false
1592986915752	Marionette	DEBUG	Setting recommended pref browser.tabs.remote.separatePrivilegedContentProcess to false
1592986915752	Marionette	DEBUG	Setting recommended pref browser.tabs.unloadOnLowMemory to false
1592986915752	Marionette	DEBUG	Setting recommended pref browser.tabs.warnOnCloseOtherTabs to false
1592986915752	Marionette	DEBUG	Setting recommended pref browser.tabs.warnOnOpen to false
1592986915753	Marionette	DEBUG	Setting recommended pref browser.usedOnWindows10.introURL to 
1592986915753	Marionette	DEBUG	Setting recommended pref browser.urlbar.suggest.searches to false
1592986915753	Marionette	DEBUG	Setting recommended pref datareporting.policy.dataSubmissionPolicyAccepted to false
1592986915753	Marionette	DEBUG	Setting recommended pref dom.disable_beforeunload to true
1592986915754	Marionette	DEBUG	Setting recommended pref dom.disable_open_during_load to false
1592986915754	Marionette	DEBUG	Setting recommended pref dom.file.createInChild to true
1592986915754	Marionette	DEBUG	Setting recommended pref dom.max_chrome_script_run_time to 0
1592986915754	Marionette	DEBUG	Setting recommended pref dom.max_script_run_time to 0
1592986915754	Marionette	DEBUG	Setting recommended pref dom.push.connection.enabled to false
1592986915754	Marionette	DEBUG	Setting recommended pref extensions.getAddons.cache.enabled to false
1592986915754	Marionette	DEBUG	Setting recommended pref extensions.getAddons.discovery.api_url to data:, 
1592986915754	Marionette	DEBUG	Setting recommended pref network.http.prompt-temp-redirect to false
1592986915754	Marionette	DEBUG	Setting recommended pref privacy.trackingprotection.enabled to false
1592986915754	Marionette	DEBUG	Setting recommended pref security.fileuri.strict_origin_policy to false
1592986915754	Marionette	DEBUG	Setting recommended pref security.notification_enable_delay to 0
1592986915755	Marionette	DEBUG	Setting recommended pref signon.autofillForms to false
1592986915755	Marionette	DEBUG	Setting recommended pref signon.rememberSignons to false
1592986915755	Marionette	DEBUG	Setting recommended pref toolkit.cosmeticAnimations.enabled to false
1592986915787	Marionette	INFO	Listening on port 33139
1592986915787	Marionette	DEBUG	Marionette is listening
1592986915799	Marionette	DEBUG	Accepted connection 0 from 127.0.0.1:47748
1592986915799	geckodriver::marionette	DEBUG	Connection to Marionette established on 127.0.0.1:33139.
1592986915799	Marionette	DEBUG	0 -> [0,1,"WebDriver:NewSession",{"acceptInsecureCerts":true,"pageLoadStrategy":"normal"}]
1592986915800	Marionette	WARN	TLS certificate errors will be ignored for this session
1592986915812	Marionette	TRACE	[6] Frame script loaded
1592986915813	Marionette	TRACE	[6] Frame script registered
1592986915816	Marionette	DEBUG	0 <- [1,1,null,{"sessionId":"03b61ab5-dfea-4a21-a5a5-fed349afffbc","capabilities":{"browserName":"firefox","browserVersion":"77.0. ... /rust_mozprofileMWB9zV","moz:shutdownTimeout":60000,"moz:useNonSpecCompliantPointerOrigin":false,"moz:webdriverClick":true}}]
1592986915817	webdriver::server	DEBUG	<- 200 OK {"value":{"sessionId":"03b61ab5-dfea-4a21-a5a5-fed349afffbc","capabilities":{"acceptInsecureCerts":true,"browserName":"firefox","browserVersion":"77.0.1","moz:accessibilityChecks":false,"moz:buildID":"20200602222727","moz:geckodriverVersion":"0.26.0","moz:headless":false,"moz:processID":30218,"moz:profile":"/tmp/rust_mozprofileMWB9zV","moz:shutdownTimeout":60000,"moz:useNonSpecCompliantPointerOrigin":false,"moz:webdriverClick":true,"pageLoadStrategy":"normal","platformName":"linux","platformVersion":"5.4.0-37-generic","rotatable":false,"setWindowRect":true,"strictFileInteractability":false,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"unhandledPromptBehavior":"dismiss and notify"}}}
2020-06-24 10:21:55.826  INFO 30131 --- [ null to remote] o.o.selenium.remote.ProtocolHandshake    : Detected dialect: W3C
2020-06-24 10:21:55.834  INFO 30131 --- [           main] c.c.selenide.webdriver.BrowserResizer    : Set browser size to 1366x768
1592986915837	webdriver::server	DEBUG	-> POST /session/03b61ab5-dfea-4a21-a5a5-fed349afffbc/window/rect {
  "width": 1366,
  "height": 768
}
1592986915847	Marionette	DEBUG	0 -> [0,2,"WebDriver:SetWindowRect",{"height":768,"width":1366}]
1592986915866	Marionette	DEBUG	0 <- [1,2,null,{"x":50,"y":0,"width":1366,"height":768}]
1592986915869	webdriver::server	DEBUG	<- 200 OK {"value":{"x":50,"y":0,"width":1366,"height":768}}
2020-06-24 10:21:55.870  INFO 30131 --- [           main] c.c.selenide.webdriver.WebDriverFactory  : BrowserName=firefox Version=77.0.1 Platform=LINUX
2020-06-24 10:21:55.870  INFO 30131 --- [           main] c.c.selenide.webdriver.WebDriverFactory  : Selenide v. 5.11.1
2020-06-24 10:21:55.870  INFO 30131 --- [           main] c.c.selenide.webdriver.WebDriverFactory  : Selenium WebDriver v. 3.141.59 build time: 2018-11-14T08:17:03
2020-06-24 10:21:55.870  INFO 30131 --- [           main] c.c.s.d.CreateDriverCommand              : Create webdriver in current thread 1: FirefoxDriver -> FirefoxDriver: firefox on LINUX (03b61ab5-dfea-4a21-a5a5-fed349afffbc)
1592986915873	webdriver::server	DEBUG	-> POST /session/03b61ab5-dfea-4a21-a5a5-fed349afffbc/url {
  "url": "http:\u002f\u002flocal.mysite:8080\u002fContext"
}
1592986915876	Marionette	DEBUG	0 -> [0,3,"WebDriver:Navigate",{"url":"http://local.mysite:8080/Context"}]
1592986915884	Marionette	TRACE	[6] Received DOM event beforeunload for about:blank
JavaScript error: resource://gre/modules/URIFixup.jsm, line 203: NS_ERROR_FAILURE: Should pass a non-null uri
```

then it hangs until Firefox windows is closed manually

```
1592987149683	Marionette	TRACE	Received DOM event unload for [object HTMLDocument]
1592987149693	Marionette	TRACE	Received observer notification message-manager-disconnect
1592987149695	Marionette	TRACE	[6] Received DOM event pagehide for about:blank
1592987149696	Marionette	TRACE	[6] Received DOM event unload for about:blank
1592987149698	Marionette	TRACE	[6] Received observer notification outer-window-destroyed
1592987149724	Marionette	DEBUG	0 <- [1,3,null,{"value":null}]
1592987149748	webdriver::server	DEBUG	<- 200 OK {"value":null}
1592987149751	webdriver::server	DEBUG	-> DELETE /session/03b61ab5-dfea-4a21-a5a5-fed349afffbc/cookie 
1592987149753	Marionette	DEBUG	0 -> [0,4,"WebDriver:DeleteAllCookies",{}]
1592987149753	Marionette	DEBUG	0 <- [1,4,{"error":"no such window","message":"Browsing context has been discarded","stacktrace":"WebDriverError@chrome://marionet ... t@chrome://marionette/content/server.js:249:9\n_onJSONObjectReady/<@chrome://marionette/content/transport.js:501:20\n"},null]
1592987149754	webdriver::server	DEBUG	<- 404 Not Found {"value":{"error":"no such window","message":"Browsing context has been discarded","stacktrace":"WebDriverError@chrome://marionette/content/error.js:175:5\nNoSuchWindowError@chrome://marionette/content/error.js:409:5\nassert.that/<@chrome://marionette/content/assert.js:454:13\nassert.open@chrome://marionette/content/assert.js:191:4\nGeckoDriver.prototype.deleteAllCookies@chrome://marionette/content/driver.js:2769:10\ndespatch@chrome://marionette/content/server.js:305:40\nexecute@chrome://marionette/content/server.js:275:16\nonPacket/<@chrome://marionette/content/server.js:248:20\nonPacket@chrome://marionette/content/server.js:249:9\n_onJSONObjectReady/<@chrome://marionette/content/transport.js:501:20\n"}}
1592987149759	Marionette	DEBUG	Closed connection 0


org.openqa.selenium.NoSuchWindowException: Browsing context has been discarded
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'francogu-linux-7540', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '5.4.0-37-generic', java.version: '1.8.0_252'
Driver info: org.openqa.selenium.firefox.FirefoxDriver
Capabilities {acceptInsecureCerts: true, browserName: firefox, browserVersion: 77.0.1, javascriptEnabled: true, moz:accessibilityChecks: false, moz:buildID: 20200602222727, moz:geckodriverVersion: 0.26.0, moz:headless: false, moz:processID: 30218, moz:profile: /tmp/rust_mozprofileMWB9zV, moz:shutdownTimeout: 60000, moz:useNonSpecCompliantPointerOrigin: false, moz:webdriverClick: true, pageLoadStrategy: normal, platform: LINUX, platformName: LINUX, platformVersion: 5.4.0-37-generic, rotatable: false, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify}
Session ID: 03b61ab5-dfea-4a21-a5a5-fed349afffbc
```

**Comment by whimboo** [comment]
@francogu does the same still happen in [Firefox Nightly](https://www.mozilla.org/en-US/firefox/channel/desktop/#nightly)?  Are there other URLs where this happens, and which are publicly accessible?

**Comment by francogu** [comment]
@whimboo Unfortunately, it seems I'm no more able to reproduce the issue despite the fact that apparently I still use the same version of the geckodriver (0.26) and Firefox (77.0.1). Previously, I was able to get 100% this error.

Since yesterday, I'm able to make it work every time. In the meantime, I performed an update of the Operating System packages. Probably, it is link to the resolution at my side.
